CorsixTH/Lua/entities/humanoids/patient.lua

```diff
@@ -356,27 +364,22 @@ function Patient:canPeeOrPuke(current)
   return parcel ~= 0 and th:getPlotOwner(parcel) == self.hospital:getPlayerIndex()
 end
 
---! Animations for when there is an earth quake
-function Patient:falling()
+--! Falling animations for when there is an earth quake or the player is being very mean
+--!param player_init (bool) if true, player triggered the fall
+function Patient:falling(player_init)
   local current = self:getCurrentAction()
   current.keep_reserved = true
   if self.falling_anim and self:canPeeOrPuke(current) and self.has_fallen == 1 then
-    self:setNextAction(FallingAction(), 0)
     self.has_fallen = 2
-    if self.has_fallen == 2 then
-      self:setNextAction(OnGroundAction())
-      self.on_ground = true
-    end
-    if self.on_ground then
-      self:setNextAction(GetUpAction())
-    end
-    -- TODO: falls are broken, the will change once implemented
-    self:interruptAndRequeueAction(current, 2)
-    self.on_ground = false
-    if math.random(1, 5) == 3 then
-      self:shakeFist()
+    self:queueAction(FallingAction(), 1)
+    self:queueAction(OnGroundAction(), 2)
+    self:queueAction(GetUpAction(), 3)
+    if math.random(1, 5) == 3 and self:shakeFist() then
+      self:interruptAndRequeueAction(current, 5)
+    else
+      self:interruptAndRequeueAction(current, 4)
     end
-    self:fallingAnnounce()
+    if player_init then self:fallingAnnounce() end
     self:changeAttribute("happiness", -0.05) -- falling makes you very unhappy
   else
     return
```

Comment:
Invert the condition so the return can be done immediately after deciding it's not happening.

Reply:
`if not self.falling_anim or not self:canPeeOrPuke(current) and not self.has_fallen == 1 then return end`?

Reply:
Since the function ends there as well, just delete the "else return" instead?

Reply:
Instead of your long condition, you can break it into 3 parts, and bail out as soon as one fails.

Reply:
An unsure why it needs a "pee or puke" function. Probably it was re-used without adapting the name.

Reply:
I think there's 3 options listed here now. Which one is the preferred one here? 😅 

> An unsure why it needs a "pee or puke" function. Probably it was re-used without adapting the name.

Very old code :) https://github.com/CorsixTH/CorsixTH/commit/df84d21f926a5868f86904b7d968bdf76d20cba6

Reply:
> Which one is the preferred one here?

Hmm, you're asking such difficult questions :)
I'd probably go for deleting the `else return`, as it's not doing anything useful.

> Very old code :)

Point is that it's being used for something else than pee or puke. The name of the function doesn't cover all it's purposes anymore. You can extend it with the new purpose, but maybe a more general name is better.? I pondered about a name and came up with `atFreeTile` or `atOpenTile`. Not sure these are good names though.

Reply:
The function checks more than tiles, it has 3 steps:
1. Is the current action interruptible?
2. Does the patient have going home status?
3. Finally check if the current tile is buildable, owned, and in a hospital

It's likely the extend name option is going to be the easiest here; can't think of a good general name either.

Reply:
I am fine with extending the name.

Good point, so perhaps this is not just 1 function, it's 2 or 3 functions mixed into one?  
The first 2 steps seem related to the question "is it possible and does it make sense to insert a new action in the queue?", while the 3rd step seems about an empty tile (which probably should be abstracted into an `isAtEmptyTile` function, but that is out of scope here).

I am not sure about the "inHospital" check for the current usage of this function. For pee and puke it makes sense, since a handyman needs to cleanup the mess, but falling doesn't need to be cleaned. Not a serious problem though, falling patients at the middle of some road are never seen anyway.  
If you want to fix things in that function, doing that in a separate patch is likely the better idea, imho.